gammapy/data/observations.py

```diff
@@ -73,11 +73,28 @@ def __init__(
         self.edisp = edisp
         self.psf = psf
         self.bkg = bkg
-        self.rad_max = rad_max
+        self._rad_max = rad_max
         self._gti = gti
         self._events = events
         self.obs_filter = obs_filter or ObservationFilter()
 
+
+    @property
+    def rad_max(self):
+        # prevent circular import
+        from gammapy.irf import RadMax2D
+
+        if self._rad_max is not None:
+            return self._rad_max
+
+        if self.aeff is not None and self.aeff.is_pointlike:
```

Comment:
Two short comments on performance / caching here:
- The access to `self.aeff` actually calls into the lazy loading interface, which by default does not cache the object. SO the repeated call to `self.aeff` will actually related the IRF from file everytime. I think the solution here is to just introduce a local variable at the top to only load the IRF once. The same is true for `edisp`
- Once could cache the created `RadMax2D` in `_rad_max` to prevent it is recreated on a 2nd access

I haven't checked which one of the mentioned points are actually relevant for the performance in case the `rad_max` use during data reduction. But might be good to keep in mind, once we start using it.

Reply:
> The access to self.aeff actually calls into the lazy loading interface, which by default does not cache the object. So the repeated call to self.aeff will actually reload the IRF from file everytime.

I thought just checking `self.aeff is None` only verifies that there is indeed a `LazyHDU` and does not load it. Is that wrong?

Reply:
Ah, sure. It calls `get`, which loads it..

Reply:
Ok, updated to cache the `_rad_max` and only load the `aeff` and `edisp` once